src/Glpi/Form/ServiceCatalog/ItemRequest.php

```diff
@@ -48,7 +50,13 @@ public function __construct(
         public int $items_per_page = ServiceCatalogManager::ITEMS_PER_PAGE,
         public SortStrategyEnum $sort_strategy = SortStrategyEnum::POPULARITY,
         public ItemRequestContext $context = ItemRequestContext::SERVICE_CATALOG,
-    ) {}
+        public ?array $form_entity_restriction = null,
+    ) {
+        // Calculate form entity restriction once during construction to avoid repeated calls
+        if ($this->form_entity_restriction === null) {
+            $this->form_entity_restriction = $this->calculateFormEntityRestriction();
+        }
+    }
```

Comment:
I don't like that we are doing it in the Item_Request because we add something specific to the FormProvider inside a generic object that will be reused by all providers.

For example, Providers added by plugins will not have this option as they can't add data to the item request.

Could you try a solution that only modify the FormProvider class itself ? Maybe it could store some kind of memory cache in a class property.